node/proof_service_factory.go

```diff
@@ -0,0 +1,39 @@
+package node
+
+import (
+	"github.com/celestiaorg/go-fraud"
+	"github.com/celestiaorg/go-fraud/fraudserv"
+	"github.com/ipfs/go-datastore"
+
+	"github.com/rollkit/rollkit/p2p"
+)
+
+type ProofServiceFactory struct {
+	client        *p2p.Client
+	getter        fraud.HeaderFetcher
+	ds            datastore.Datastore
+	syncerEnabled bool
+	proofType     fraud.ProofType
+}
+
+func NewProofServiceFactory(c *p2p.Client, getter fraud.HeaderFetcher, ds datastore.Datastore, syncerEnabled bool, proofType fraud.ProofType) ProofServiceFactory {
+	return ProofServiceFactory{
+		client:        c,
+		getter:        getter,
+		ds:            ds,
+		syncerEnabled: syncerEnabled,
+		proofType:     proofType,
+	}
+}
+
+// OnStart is a part of Service interface.
+func (factory *ProofServiceFactory) Start() *fraudserv.ProofService {
```

Comment:
[optional, nitpicking] calling `factory.Start()` looks like starting factory, not instantiating object by the factory.
`Create` seems more natural.

Reply:
Agreed since `Start` is not part of the `Service` interface. Maybe even `CreateProofService` if we want to be more explicit.

Reply:
agreed.